alwaysontop/render.js

```diff
@@ -244,20 +277,28 @@ class AlwaysOnTop {
      */
     _closeAlwaysOnTopWindow() {
         if (this._alwaysOnTopBrowserWindow) {
-            const position
-                = this._alwaysOnTopBrowserWindow.getPosition();
-            this._alwaysOnTopBrowserWindow = undefined;
-            this._position = {
-                x: position[0],
-                y: position[1]
-            };
+            if (!this._alwaysOnTopBrowserWindow.isDestroyed()) {
+                const position =
+                    this._alwaysOnTopBrowserWindow.getPosition();
+
+                this._position = {
+                    x: position[0],
+                    y: position[1]
+                };
+            }
         }
+
         if(this._alwaysOnTopWindow) {
-            this._alwaysOnTopWindow.close();
-            this._alwaysOnTopWindow = undefined;
+            if(!this._alwaysOnTopBrowserWindow ||!this._alwaysOnTopBrowserWindow.isDestroyed()) {
```

Comment:
Do we need this checks here? Why?

Reply:
there is a problem with timing in electron if you try to close a window multiple times in a row. if you issue two calls to `.close()` on the javascript window object, the second call will throw a RPC exception. Also, the window.closed property is not set to true immediately after the first call so there is no reliable way of preventing the exception. Checking BrowserWindow.isDestroyed() is robust though.

Reply:
oh, I misread the question... experimentally, yes, I guess there is some problem with how those references are being handles, but that is a separate, pre-existing issue.

Reply:
Experimentally? How? Could you please elaborate?

Reply:
updated :D